src/span_context.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as constants from './constants.js';
+import Utils from './util.js';
+import Int64 from 'node-int64';
+
+export default class SpanContext {
+    _traceId: any;
+    _spanId: any;
+    _parentId: any;
+    _flags: number;
+    _baggage: any;
+    _baggageHeaderCache: any;
+
+    constructor(traceId: any,
+                spanId: any,
+                parentId: any,
+                flags: number) {
+        this._traceId = traceId;
+        this._spanId = spanId;
+        this._parentId = parentId;
+        this._flags = flags;
+        this._baggage = {};
+        this._baggageHeaderCache = {};
+    }
+
+    get traceId(): any {
+        return this._traceId;
+    }
+
+    get spanId(): any {
+        return this._spanId;
+    }
+
+    get parentId(): any {
+        return this._parentId;
+    }
+
+    get flags(): number {
+        return this._flags;
+    }
+
+    get baggage(): any {
+        return this._baggage;
+    }
+
+    _normalizeBaggageKey(key: string) {
+        if (key in this._baggageHeaderCache) {
+            return this._baggageHeaderCache[key];
+        }
+
+        // This stackoverflow claims this approach is slower than regex
+        // http://stackoverflow.com/questions/1144783/replacing-all-occurrences-of-a-string-in-javascript
+        let normalizedKey: string = key.toLowerCase().split('_').join('-');
+        // jshint validthis:false
+        //
+        // jshint validthis:true
+        if (Object.keys(this._baggageHeaderCache).length < 100) {
+            this._baggageHeaderCache[key] = normalizedKey;
+        }
+        // jshint validthis:false
+
+        return normalizedKey;
+    }
+
+    setBaggageItem(key: string, value: string): void {
```

Comment:
add todo to move these to Span, looks like JS API has not been upgraded to immutable Span Context. I asked ben and bcronin if they're planning to do that, if not we may need to do a PR to OT, as I do not want to release Node client that needs to change the API again shortly.